jpa/subsystem/src/main/java/org/jboss/as/jpa/classloader/JPADelegatingClassFileTransformer.java

```diff
@@ -52,30 +57,40 @@ public JPADelegatingClassFileTransformer(PersistenceUnitMetadata pu) {
     @Override
     public ByteBuffer transform(ClassLoader classLoader, String className, ProtectionDomain protectionDomain, ByteBuffer classBytes)
```

Comment:
Transform() method was changed to wrap as security manager checked code.

Reply:
Is this needed for this bug?

Reply:
Yes, we are now seeing later calls to transform application entity classes than we saw without this change.  When we run the new unit test with the Java security manager, we fail without this change.

Reply:
Can this be moved to the same package as JPAClassFileTransformerProcessor and the class made package-protected instead of public? If this needs to be done with a privileged block we should lock down the ability for unknown code to create one of these.

Reply:
@bstansberry done